layers/+distributions/spacemacs-base/funcs.el

```diff
@@ -184,6 +184,49 @@ Dedicated (locked) windows are left untouched."
   (interactive "p")
   (spacemacs/rotate-windows (* -1 count)))
 
+(defun spacemacs/move-buffer-to-window (windownum follow-focus-p)
+  "Moves a buffer to a window, using the spacemacs numbering. follow-focus-p
+   controls whether focus moves to new window (with buffer), or stays on
+   current"
+  (interactive)
+  (let ((b (current-buffer))
+        (w (selected-window)))
+    (save-selected-window
+      (select-window-by-number windownum)
+      (switch-to-buffer b)
+      (select-window w)
+      (switch-to-prev-buffer)
+      (unrecord-window-buffer (selected-window) b)))
+  (if follow-focus-p (select-window-by-number windownum)))
+
+(defun spacemacs/swap-buffers-to-window (windownum follow-focus-p)
+  "Swaps visible buffers between active window and selected window.
+   follow-focus-p controls whether focus moves to new window (with buffer), or
+   stays on current"
+  (interactive)
+  (let ((b1 (current-buffer))
+        (w1 (selected-window)))
+    (save-selected-window
+      (select-window-by-number windownum)
+      (let ((b2 (current-buffer))
+            (w2 (selected-window)))
+        (unless (eq w1 w2)
+          (switch-to-buffer b1)
+          (unrecord-window-buffer w2 b2)
+          (select-window w1)
+          (switch-to-buffer b2)
+          (unrecord-window-buffer w1 b1)))))
+  (if follow-focus-p (select-window-by-number windownum)))
```

Comment:
When there is no `else` clause, `when` is better than `if` (both semantically and because it allows more than on sexp in the `then` clause).